phpBB/posting.php

```diff
@@ -606,9 +606,24 @@
 }
 unset($uninit);
 
+$attachment_user_id = $post_data['poster_id'];
+/**
+* This event allows you to modify the poster_id used in get_submitted_attachment_data
+*
+* @event core.posting_modify_attachment_user_id
+* @var	array	post_data	Array with post data
+* @var	int		attachment_user_id	Poster ID used to get attachment data
+* @since 3.2.5-RC1
+*/
+$vars = array(
+	'post_data',
+	'attachment_user_id',
+);
+extract($phpbb_dispatcher->trigger_event('core.posting_modify_attachment_user_id', compact($vars)));
+
 // Always check if the submitted attachment data is valid and belongs to the user.
 // Further down (especially in submit_post()) we do not check this again.
-$message_parser->get_submitted_attachment_data($post_data['poster_id']);
```

Comment:
I'd move the event underneath this line and add the `message_parser` to the array.

Then undo the changes to the `message_parser` 's line of code.

Reply:
Unfortunately if `get_submitted_attachment_data` is called at all with an unmodified poster_id in my situation, users would still get an error.  I attempted to move it and add the message parser and re execute it under my conditions, it doesn't bypass the error.

Reply:
Without testing I'd go that way, more or less.

```
$message_parser->get_submitted_attachment_data($post_data['poster_id']);

/**
* This event allows you to modify the post data etc..
*
* @event core.posting_modify_post_data
* @var	array	post_data		Array with post data
* @var	@todo	message_parser	The message parser etc..
* @since 3.2.5-RC1
*/
$vars = array(
	'post_data',
	'message_parser',
);
extract($phpbb_dispatcher->trigger_event('core.posting_modify_post_data', compact($vars)));

```

and in the event
```

$message_parser = event['message_parser'];
$poster_id = event['post_data']['poster_id'];

$my_id = do something to define your user ID;
$poster_id = (int) $my_id;

$message_parser->get_submitted_attachment_data($poster_id);

event['message_parser'] = event['message_parser'];

```

Just to throw something on the table though.

Reply:
That's pretty much exactly what I did after your initial suggestion.  Since `$message_parser->get_submitted_attachment_data($post_data['poster_id'])` is ran with the wrong poster_id initially, it throws the error and doesn't give the message_parser a second chance to run.

Reply:
![image](https://user-images.githubusercontent.com/29783193/48930199-e8b15c00-eebc-11e8-8f44-87ac416731d0.png)

See the error, it gets automatically called at line 614 (the first call to get_submitted_attachment_data), and nothing in the event can be executed.

Reply:
Reading the code a bit more, the `poster_id` is being set here (` $user->data['user_id']` )

https://github.com/phpbb/phpbb/blob/838b5159acf3a10168b28c23226ff229b7960c90/phpBB/posting.php#L597

Why don't you try to use that array  `uninit` with a new event?

Reply:
I'll test doing that tomorrow, as I'm unsure of the affect of letting the remainder of posting.php process the post with a modified poster_id.  I'd like to keep it ANONYMOUS as long as I can, until I might change it in other events.  Nevertheless I'll attempt this tomorrow and see what happens.

Reply:
Seems to be working just fine, and I suppose this event is more useful than the one I made anyways.  Thanks for your suggestions and help.